app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -480,16 +480,32 @@ def process_disks(vm)
 
     def process_vm_guest_devices(vm)
       devices = []
-      return devices if vm['VirtualDVDDrives'].blank?
 
-      vm['VirtualDVDDrives'].each do |dvd|
-        devices.concat(process_vm_physical_dvd_drive(dvd)) unless dvd['HostDrive'].blank?
+      vm['VirtualNetworkAdapters'].to_a.each do |vnic|
+        devices << process_vm_virtual_nic(vnic)
+      end
+
+      vm['VirtualDVDDrives'].to_a.each do |dvd|
+        devices << process_vm_physical_dvd_drive(dvd) unless dvd['HostDrive'].blank?
```

Comment:
Unrelated to this BZ but I changed this from `devices.concat()` to `devices <<` because `process_vm_physical_dvd_drive` is returning a hash not an array so I don't see how this would have worked if `dvd['HostDrive']` wasn't nil.